packages/itwinui-react/src/core/Table/Table.tsx

```diff
@@ -748,11 +748,34 @@ export const Table = <
     ],
   );
 
+  const headerRef = React.useRef<HTMLDivElement>(null);
+  const bodyRef = React.useRef<HTMLDivElement>(null);
+
   const { scrollToIndex, tableRowRef } = useScrollToRow<T>({ ...props, page });
   const columnRefs = React.useRef<Record<string, HTMLDivElement>>({});
   const previousTableWidth = React.useRef(0);
   const onTableResize = React.useCallback(
     ({ width }: DOMRectReadOnly) => {
+      // ------------------------------------------------------------------------------------------------
+      // Handle setShouldShowDummyDiv
+
+      // Show the dummy div only if …
+      setShouldShowDummyDiv(
+        // … data is empty
+        data.length === 0 &&
+          // … and header is indeed overflowing
+          (headerRef.current != null
+            ? headerRef.current?.scrollWidth > headerRef.current.clientWidth
+            : false),
+      );
+
+      // ------------------------------------------------------------------------------------------------
+      // Handle table properties, but only when table is resizable
+
+      if (!isResizable) {
+        return;
+      }
```

Comment:
Instead of changing the state inside `onTableResize`, would it be sufficient to just do something like this?

```tsx
<ShadowTemplate>
    <slot />
    {data.length === 0 && headerRef.current (
      <div
        aria-hidden
        style={{
          width:
            headerRef.current.scrollWidth > headerRef.current.clientWidth
              ? headerRef.current.scrollWidth
              : 0,
          height: 0.1
        }}
    )}
</ShadowTemplate>
```

Reply:
I made a few changes to fix some errors:

```diff
<ShadowTemplate>
    <slot />
-     {data.length === 0 && headerRef.current (
+     {data.length === 0 && headerRef.current && (
      <div
        aria-hidden
        style={{
          width:
-             headerRef.current.scrollWidth > headerRef.current.clientWidth
+            headerRef.current != null && headerRef.current.scrollWidth > headerRef.current.clientWidth
              ? headerRef.current.scrollWidth
              : 0,
          height: 0.1
        }}
      />
    )}
</ShadowTemplate>
```

---

I had tried something similar to this approach. But the problem with that is it does not re-run the logic when the `scrollWidth` and `clientWidth` changes.

So, when the table/column width changes, the dummy div's width does not update accordingly.

Reply:
Sounds good, I stopped using `ref.current` in renders. So now I created two state variables (`headerScrollWidth` & `headerClientWidth`) that are updated in `onTableResize` (https://github.com/iTwin/iTwinUI/pull/1809/commits/b6ac9efac5e54720442cc30882b4fb053f0bd228).